packages/protocol/test/identity/attestationsubsidy.ts

```diff
@@ -0,0 +1,307 @@
+const MockStableToken: MockStableTokenContract = artifacts.require('MockStableToken')
+const MockElection: MockElectionContract = artifacts.require('MockElection')
+const MockLockedGold: MockLockedGoldContract = artifacts.require('MockLockedGold')
+const MockValidators: MockValidatorsContract = artifacts.require('MockValidators')
+const Random: MockRandomContract = artifacts.require('MockRandom')
+const Registry: RegistryContract = artifacts.require('Registry')
+const AttestationSubsidy: AttestationSubsidyContract = artifacts.require('AttestationSubsidy')
+const MetaTransactionWallet: MetaTransactionWalletContract = artifacts.require(
+  'MetaTransactionWallet'
+)
+
+const getSignatureForMetaTx = async (
+  walletAddress: Address,
+  signer: Address,
+  tx: MetaTransaction
+) => {
+  const digest = constructMetaTransactionExecutionDigest(walletAddress, tx)
+  return getSignatureForDigest(digest, signer)
+}
+
+contract('AttestationSubsidy', (accounts: string[]) => {
+  let attestationSubsidy: AttestationSubsidyInstance
+  let accountsInstance: AccountsInstance
+  let attestations: TestAttestationsInstance
+  let mockStableToken: MockStableTokenInstance
+  let otherMockStableToken: MockStableTokenInstance
+  let random: MockRandomInstance
+  let mockElection: MockElectionInstance
+  let mockLockedGold: MockLockedGoldInstance
+  let registry: RegistryInstance
+  let callerMetaWallet: MetaTransactionWalletInstance
+
+  const provider = new Web3Class.providers.HttpProvider('http://localhost:8545')
+  const web3: Web3 = new Web3Class(provider)
+  const phoneNumber: string = '+18005551212'
+  const subsidyOwner: string = accounts[0]
+  const caller: string = accounts[1]
+  // Private keys of each of the 10 miners, in the same order as their addresses in 'accounts'.
+  const accountPrivateKeys: string[] = [
+    '0xf2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e0164837257d',
+    '0x5d862464fe9303452126c8bc94274b8c5f9874cbd219789b3eb2128075a76f72',
+    '0xdf02719c4df8b9b8ac7f551fcb5d9ef48fa27eef7a66453879f4d8fdc6e78fb1',
+    '0xff12e391b79415e941a94de3bf3a9aee577aed0731e297d5cfa0b8a1e02fa1d0',
+    '0x752dd9cf65e68cfaba7d60225cbdbc1f4729dd5e5507def72815ed0d8abc6249',
+    '0xefb595a0178eb79a8df953f87c5148402a224cdf725e88c0146727c6aceadccd',
+    '0x83c6d2cc5ddcf9711a6d59b417dc20eb48afd58d45290099e5987e3d768f328f',
+    '0xbb2d3f7c9583780a7d3904a2f55d792707c345f21de1bacb2d389934d82796b2',
+    '0xb2fd4d29c1390b71b8795ae81196bfd60293adf99f9d32a0aff06288fcdac55f',
+    '0x23cb7121166b9a2f93ae0b7c05bde02eae50d64449b2cbb42bc84e9d38d6cc89',
+  ]
+
+  const phoneHash: string = getPhoneHash(phoneNumber)
+
+  const attestationsRequested = 3
+  const attestationExpiryBlocks = (60 * 60) / 5
+  const selectIssuersWaitBlocks = 4
+  const maxAttestations = 20
+  const attestationFee = new BigNumber(web3.utils.toWei('.05', 'ether').toString())
+
+  // async function getVerificationCodeSignature(
```

Comment:
Remember to remove commented out code.

Reply:
Yeah tests are WIP didn't get any to pass yet and I copied the attestations test setup as a starting point and I'm not yet sure what I will actually need from that and what I won't, still wrapping my head around that test setup.